### PR TITLE
do_not_disturb module: fix startup notif and pause logic

### PR DESCRIPTION
**Pull request by ultrabug**

### CODE REVIEW

**Comment by lasers** [comment]
>I'd rather drop the pause setting and only rely on USR1 / USR2 signals

Sure... as long as you know the history, #1272. @maximbaz and others might prefer it.

Care to explain more about wrong pause logic? DND's `0/1` state is opposite and confusing if we compare our pause/resume to doc's pause/resume.... but it looks to be doing the right thing.

Assuming `dunst` added paused later to deal with i3lock, notification history, we can't go wrong with defaulting either option.

File: py3status/modules/do_not_disturb.py [hunk 1]
```diff
@@ -112,14 +109,17 @@ def toggle(self, state):
         self.iface.SetProperty("xfce4-notifyd", "/do-not-disturb", state)
 
 
+class Meta:
+    deprecated = {"remove": [{"param": "pause", "msg": "obsolete parameter"}]}
```
Comment (maximbaz) [inline py3status/modules/do_not_disturb.py]: Make sure to update `Dunst Miscellaneous:` section in the docstring in the top of the file

**Comment by maximbaz** [comment]
I would indeed prefer to use the old `pause=False` behavior, so that when I disable DND I don't get spammed with a gazillion of notifications. Why is this change needed, to resolve conflict with weather module? Maybe changing the default value of `pause` option would be enough?

**Comment by maximbaz** [comment]
@lasers could you explain what exactly is broken on `master`, if anything? I've just tested, everything seems to be working perfectly fine, but maybe I'm missing some particular scenario.

**Comment by lasers** [comment]
Add `order += "weather_yahoo"` to your config. Without `DND`, py3 sends a notification about the obsolete module. With `DND`, it does that for like 0.1 seconds then disappear because `dunst` just got a signal from `DND` module.

I'm not sure if it's the module at issue here because if that happens, then I'm assuming it was a `SIGUSR1` signal that causes `dunst` to sometimes disappear with DND state off.

**Comment by maximbaz** [comment]
added "weather_yahoo".

With dunst and DND from master:

- notifications allowed, restart i3: getting 'module is no longer available' 👍 
- notifications disabled, restart i3: not getting anything 👍 
    - then click on DND to allow notifications: nothing will appear 👍 

adding `pause = True`:

- notifications allowed, restart i3: getting 'module is no longer available' 👍 
- notifications disabled, restart i3: not getting anything 👍 
    - then click on DND to allow notifications: 'module is no longer available' 👍 


What am I missing? Seems all is good?

**Comment by lasers** [comment]
I think maybe dunst sometimes glitches or act up because of i3 restarts or other things. For most part, everything looks okay here on my side, but maybe it wasn't setting a `DND` state fast enough or dunst didn't have enough time to deal with the first notification... so users could occasionally see a notification for a very short time.

@ultrabug We could try `sleep` for a second at end of `post_config_hook` to give dunst some time... if the issue above was your issue.

**Comment by maximbaz** [comment]
I think that would be a bug in dunst, because the code in master doesn't touch `dunst` process on i3 restart, it doesn't send any signal to dunst (at least with the default  config where `state` is `"last"`). So if dunst promised that it is paused and then flickers without any action from our side, it's dunst alone to blame.

Shall we close this PR then and keep code in master as it is?

**Comment by lasers** [comment]
If `state` is `True`, `False`, or `'last'`, then it will get toggled with that state at startup + i3 restarts.
https://github.com/ultrabug/py3status/blob/590820548f4bdfda0f57c86eba10f49964e6fe99/py3status/modules/do_not_disturb.py#L156-L162

**Comment by maximbaz** [comment]
In your snippet the call to `toggle` is in `elif` block, meaning it is *not* getting executed when `self.state == "last"`. So by default, `toggle` is not executed. Right? :)

**Comment by lasers** [comment]
Right... as in... it just got changed 7 hours ago breaking `last` state.

**Comment by maximbaz** [comment]
Can you tell me how to reproduce what exactly is broken?

**Comment by lasers** [comment]
@maximbaz It's a race condition. Some restarts works better. Some restarts causes issues.

**Comment by ultrabug** [comment]
real fix is on #1624

**Pull request closed without merging**